src/de.rs

```diff
@@ -308,7 +308,11 @@ impl<'de> de::SeqAccess<'de> for PySequenceAccess<'de> {
         T: de::DeserializeSeed<'de>,
     {
         if self.index < self.len {
-            let mut item_de = Depythonizer::from_object(self.seq.get_item(self.index)?);
+            let item = match self.seq.next() {
+                Some(item) => item?,
+                None => return Ok(None),
```

Comment:
This branch shouldn't happen for `set` or `frozenset`, since the length is known.

It might not be necessary to have the index checks any more, as long as we still check the `length` is finite.